scipy/stats/_binned_statistic.py

```diff
@@ -583,12 +583,13 @@ def binned_statistic_dd(sample, values, statistic='mean',
             result[vv, a] = flatsum[a] / flatcount[a]
     elif statistic == 'std':
         result.fill(0)
-        flatcount = np.bincount(binnumbers, None)
-        a = flatcount.nonzero()
+        unique_bin_numbers = np.unique(binnumbers)
         for vv in builtins.range(Vdim):
-            for i in np.unique(binnumbers):
+            bin_map = _create_bin_map(binnumbers, unique_bin_numbers, values,
```

Comment:
the signature/naming here is weird, `binnumbers` and `unique_bin_numbers` shouldn't both be necessary information, is there a reason I'm missing?

Reply:
I want to reduce unnecessary calculation as much as possible, so I just call `unique` just one time at outer loop, and reuse it in the `Vdim` for loop. That's why I pass `binnumbers` and `unique_bin_numbers` to the` _create_bin_map` function. I